### PR TITLE
Add function of creating route titile and add carriage's link to train

### PR DESCRIPTION
**Pull request by mersXX**
Add callbacks, new carriage types, and ability to create routes using existing stations.

### CODE REVIEW

File: app/controllers/carriages_controller.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@ def create
     @carriage = Carriage.new(carriage_params)
 
     if @carriage.save
-      redirect_to @carriage
```
Comment (kavalenka) [inline app/controllers/carriages_controller.rb]: Please add .idea/workspace.xml to .gitignore and untrack this file.

File: app/controllers/carriages_controller.rb [hunk 2]
```diff
@@ -45,6 +45,14 @@ def set_carriage
...
-    params.require(:carriage).permit(:carriage_type, :top_seats, :bottom_seats, :train_id)
+    params.require(:carriage).permit(:carriage_type,
+                                     :type,
+                                     :top_seats,
+                                     :bottom_seats,
+                                     :side_bottom_seats,
+                                     :side_top_seats,
+                                     :sitting_seats,
+                                     :number,
+                                     :train_id,)
```
Comment (kavalenka) [inline app/controllers/carriages_controller.rb]: Remove extra comma

File: app/models/coupe_carriage.rb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+class CoupeCarriage < Carriage
+
```
Comment (kavalenka) [inline app/models/coupe_carriage.rb]: Please delete empty row

File: app/models/econom_carriage.rb [hunk 4]
```diff
@@ -0,0 +1,3 @@
+class EconomCarriage < Carriage
+
```
Comment (kavalenka) [inline app/models/econom_carriage.rb]: One more here

File: app/models/train.rb [hunk 5]
```diff
@@ -4,5 +4,18 @@ class Train < ApplicationRecord
   belongs_to :route
   has_many :tickets
   has_many :carriages
+
+  before_validation :set_number
+
+  private
+
+  def set_number
+    self.number = Train.all.count + 1
```
Comment (kavalenka) [inline app/models/train.rb]: No need to use method .all here

File: app/views/carriages/show.html.erb [hunk 6]
```diff
@@ -1,9 +1,13 @@
 <p id="notice"><%= notice %></p>
 
 <p>
-  <strong>Type:</strong>
-  <%= @carriage.carriage_type %>
-  <br>
+  <strong>These carriage:</strong>
+  <% if @carriage.type.eql?('EconomCarriage') %>
+    <%= render 'econom_carriage', carriage: @carriage %>
+    <% else @carriage.type.eql?('CoupeCarriage') %>
```
Comment (kavalenka) [inline app/views/carriages/show.html.erb]: It's better to use ``elsif`` with another condition here. Use ``else`` only with all other uncovered conditions.

**Comment by kavalenka** [comment]
Also, what does it mean ``Rsr index`` ?
Please use more meaningful naming.

File: app/controllers/carriages_controller.rb (hunk 1 above)
Comment (mersXX) [reply app/controllers/carriages_controller.rb]: file .idea/workspace.xml already added to .gitignore

**Pull request merged**